chrome/content/api/FPVS/implementation.js

```diff
@@ -0,0 +1,394 @@
+                )
+              };
+            } catch (e) { }
+
+            prefs.prefs = {};
+
+            let children = viewsBranch.getChildList(""); //, obj);
+            log("children", children);
+            let regex = /^(\d+)\./;
+            for (let child of children) {
+              let match = regex.exec(child);
+              let num = match[1];
+              let name = viewsBranch.getStringPref(num + ".name");
+              let arrow = viewsBranch.getBoolPref(
+                num + ".arrows_enabled"
+              );
+              let menu = viewsBranch.getBoolPref(
+                num + ".menu_enabled"
+              );
+
+              //       if (["all", "smart", "recent", "favorite", "unread"].includes (name) )  prefs.prefs[name] = {"arrow": arrow, "menu": menu, "pos": -1};
+              if (allViews.includes(name))
+                prefs.prefs[name] = {
+                  arrow: arrow,
+                  menu: menu,
+                  pos: -1
+                };
+            }
+
+            log("prefs", prefs);
+
+            return prefs;
+            /*        */
+            //       return "prefs";
+          },
+
+          getViewDisplayName: async function (windowId, commonName) {
+            let mail3Pane = context.extension.windowManager.get(windowId).window;
+            let key = "folderPaneModeHeader_" + commonName;
+            let nameString =
+              mail3Pane.gFolderTreeView.messengerBundle.getString(
+                key
+              );
+            log("legname", nameString);
+            return nameString;
+          },
+
+          showViewInMenus: async function (windowId, view, enabled) {
+            log("showViewInMenus");
+            let mail3Pane = context.extension.windowManager.get(windowId).window;
+            let item = mail3Pane.document.querySelector(
+              `#folderPaneOptionsPopup [value=${view}]`
+            );
+            if (item != null) {
+              item.hidden = !enabled;
+            }
+            item = mail3Pane.document.querySelector(
+              `#menu_FolderViewsPopup [value=${view}]`
+            );
+            if (item != null) {
+              item.hidden = !enabled;
+            }
+            item = mail3Pane.document.querySelector(
+              `#appMenu-foldersView [value=${view}]`
+            );
+            if (item != null) {
+              item.setAttribute("hidden", !enabled);
+            }
+          },
+
+          toggleElementHidden: async function (windowId, should_be_hidden) {
+            let mail3Pane = context.extension.windowManager.get(windowId).window;
+            let is_hidden = !!mail3Pane.document.getElementById(
+              "FolderPaneSwitcher-back-arrow-button"
+            ).hidden;
+            if (should_be_hidden != is_hidden) {
+              mail3Pane.document.getElementById(
+                "FolderPaneSwitcher-back-arrow-button"
+              ).hidden = should_be_hidden;
```

Comment:
Don't get a DOM node more than one time within one function. Fetch it once and use a variable.